pkg/bootstrapper/node_bootstrapper.go

```diff
@@ -10,38 +34,382 @@ package bootstrapper
+	if err != nil {
+		return []byte{}, err
 	}
+	newContents := contents.Data
+	if fileTranslateFn != nil {
+		newContents, err = fileTranslateFn(wcb, contents.Data)
+		if err != nil {
+			return []byte{}, err
+		}
+	}
+	return newContents, err
 }
 
-// parseIgnitionFile parses the ignition file and returns the contents of kubeconfig and kubelet config
-func parseIgnitionFile() (string, string) {
-	// Parses the ignition file
-	return "", ""
+// These regex are global, so that we only need to compile them once
+// cloudProviderRegex searches for the cloud provider option given to the kubelet
+var cloudProviderRegex = regexp.MustCompile(`--cloud-provider=(\w*)`)
+
+// verbosityRegex searches for the verbosity option given to the kubelet
+var verbosityRegex = regexp.MustCompile(`--v=(\w*)`)
+
+// parseIgnitionFile parses the ignition file and writes the contents of the described files
+// to the k8s installation directory
+func (wcb *winNodeBootstrapper) parseIgnitionFile(ignitionFilePath string, filesToTranslate map[string]fileTranslation) error {
+	ignitionFileContents, err := ioutil.ReadFile(ignitionFilePath)
+	if err != nil {
+		return err
+	}
+	// Parse configuration file
+	configuration, _, err := ignitionv2.Parse(ignitionFileContents)
+	if err != nil {
+		return err
+	}
+	for _, ignFile := range configuration.Storage.Files {
+		if filePair, ok := filesToTranslate[ignFile.Node.Path]; ok {
+			newContents, err := wcb.translateFile(ignFile.Contents.Source, filePair.translationFunc)
+			if err != nil {
+				return fmt.Errorf("could not process %s: %s", ignFile.Node.Path, err)
+			}
+			if err = ioutil.WriteFile(filePair.dest, newContents, 0644); err != nil {
+				return fmt.Errorf("could not write to %s: %s", filePair.dest, err)
+			}
+		}
+	}
+	for _, unit := range configuration.Systemd.Units {
+		if unit.Name == kubeletSystemdName {
+			results := cloudProviderRegex.FindStringSubmatch(unit.Contents)
+			if len(results) == 2 {
+				wcb.kubeletArgs["cloud-provider"] = results[1]
+			}
+			results = verbosityRegex.FindStringSubmatch(unit.Contents)
+			if len(results) == 2 {
+				wcb.kubeletArgs["v"] = results[1]
+			}
+		}
+
+	}
+	return nil
 }
 
 // Initializes the kubelet after copying the kubelet to the desired location
-func (wcb *winNodeBootstrapper) initializeKubelet() (bool, error) {
-	return false, nil
+func (wcb *winNodeBootstrapper) initializeKubelet() error {
+	filesToTranslate := map[string]fileTranslation{
+		"/etc/kubernetes/kubelet.conf": {
+			dest:            wcb.kubeletConfPath,
+			translationFunc: prepKubeletConfForWindows,
+		},
+		"/etc/kubernetes/kubeconfig": {
+			dest: filepath.Join(wcb.installDir, "bootstrap-kubeconfig"),
+		},
+		"/etc/kubernetes/kubelet-ca.crt": {
+			dest: filepath.Join(wcb.installDir, "kubelet-ca.crt"),
+		},
+	}
+	err := os.MkdirAll(wcb.installDir, os.ModeDir)
+	if err != nil {
+		return fmt.Errorf("could not make install directory: %s", err)
+	}
+	if wcb.initialKubeletPath != "" {
```

Comment:
Is it valid for `initialKubeletPath` to be `""`?

Reply:
Only in the case that the files already exist. As for if we want to entertain that case, I'm not sure.